radioactivedecay/nuclide.py

```diff
@@ -0,0 +1,651 @@
+            100: "Fm", 101: "Md", 102: "No", 103: "Lr", 104: "Rf", 105: "Db",
+            106: "Sg", 107: "Bh", 108: "Hs", 109: "Mt", 110: "Ds", 111: "Rg",
+            112: "Cn", 113: "Nh", 114: "Fl", 115: "Mc", 116: "Lv", 117: "Ts",
+            118: "Og"
+        }
+        sym_dict = dict((v,k) for k,v in Z_dict.items())
+        
+        self.Z_dict = Z_dict
+        self.sym_dict = sym_dict
+        self.parse_name(input, A, meta_state)
+        self.data = data
+
+    def parse_name(
+        self,
+        vary_input: Any,
+        A: int,
+        meta_state: str
+        ) -> None:
+        
+        if A == 0:
+            if type(vary_input) == int:
+                if vary_input > 10000000:
+                    id_zzzaaa = int(vary_input / 10000)
+                    state_digits = vary_input - (id_zzzaaa * 10000)
+                    if state_digits == 1:
+                        meta_state = "m"
+                    elif state_digits == 2:
+                        meta_state = "n"
+                    Z = int(id_zzzaaa / 1000)
+                    A = id_zzzaaa - (Z * 1000)
+                else:
+                    Z = vary_input
+                
+                name = self.build_nuclide_string(Z, A, meta_state)
+            elif type(vary_input) == str:
+                if any(i in vary_input for i in ["0","1","2","3","4","5","6","7","8","9"]):
+                    name = parse_nuclide(vary_input)
+                    Z = self.sym_dict[name.split("-")[0]]
+                    A = int(name.split("-")[1].strip("mn"))
+                elif vary_input in self.Z_dict.values():
+                    name = vary_input
+                    Z = self.sym_dict[vary_input]
+                else:
+                    raise ValueError(
+                        vary_input + ' is not a valid element or nuclide'
+                    )
+            else:
+                raise TypeError(
+                    "Invalid name input type, string or int expected"
+                )
+        elif type(vary_input) == int and type(A) == int:
+            Z = vary_input
+            name = self.build_nuclide_string(Z, A)
+        else:
+            raise TypeError(
+                "Invalid input types, expected int"
+            )
+        
+        self.nuclide_name = name
+        self.Z_val = Z
+        self.A_val = A
+    
+    def build_nuclide_string(
+        self,
+        Z: int,
+        A: int = 0,
+        meta_state: str = ""
+        ) -> str:
+                    
+        if Z not in self.Z_dict.keys():
+            raise ValueError(
+                str(Z) + ' is not a valid atomic number'
+            )
+            
+        return_string = self.Z_dict[Z]
+        if A != 0:
+            return_string += "-" + str(A) + meta_state
+            
+        return return_string
+    
+    def __repr__(self) -> str:
+        if self.A == 0:
+            rep =  "Element: " + self.nuclide_name
```

Comment:
Likewise this felt a little un-pythonic, how the Nuclide class can also behave like an Element class. A more natural way would be to have a separate Element class if needed. But do we need an Element class? Did you have a use case in mind when you made this?